Logger/Storage.cs

```diff
@@ -20,10 +20,8 @@ public bool Contains(IEntity item)
     
     public IEntity? Get(Guid expectedGuid)
     {
-        return Entities.FirstOrDefault(entity => 
-        {
-            dynamic dynamicEntity = entity;
-            return dynamicEntity.Id == expectedGuid;
-        });
+        return Entities
+        .OfType<IEntity>()
+        .FirstOrDefault(entity => entity.Id == expectedGuid);
```

Comment:
```suggestion
        return Entities.FirstOrDefault(entity => entity.Id == expectedGuid);
```
This can be simplified